src/app/utils/pickupTimeEstimator.js

```diff
@@ -352,18 +352,22 @@ estimator._addMinutes = (dateString, minutes) => {
  *  calculating estimates just before/after Daylight Savings days)
  */
 estimator._nyOffset = (timestamp = estimator._now()) => {
-  // TODO: This should be driven by server time, i.e.:
-  //
-  // Assume we build a nyOffsets var on the server representing the next week
-  // of offsets and make it available as:
-  //   window.nyOffsets = { '2023-01-01': 5, '2023-01-02': 5, ...}
-  // Then this function can determine the NY offset for the requested
-  // timestamp via:
-  //   const day = timestamp.split('T').shift()
-  //   const offset = window && window.nyOffsets ? window.nyOffsets[day] : new Date(timestamp).getTimezoneOffset() / 60
-
-  const offset = new Date(timestamp).getTimezoneOffset() / 60
-  return offset
+  if (window.nyOffsets && window.nyOffsets.length) {
+    // Identify the offset that starts before timestamp:
+    const currentOffset = window.nyOffsets
+      .filter((offset) => new Date(offset.from) < new Date(timestamp))
```

Comment:
I think this could be a `find`

Reply:
Find chooses first match; We want the last matching element.

A few days before DST, the `window.nyOffsets` var will be something like:
```
[
  { from: '2024-03-07T15:00:00Z', offset: 5 },
  { from: '2024-03-10T07:00:00Z', offset: 4 }
]
```
If we're calling `_nyOffset('2024-03-11T02:00:00Z')` to get the TZ offset of the day following DST, a `find` would return the first entry, when we actually want the last one (i.e. we want the last entry whose `from` is LT the timestamp).